### PR TITLE
Audio changes

### PR DESCRIPTION
**Pull request by Timoram**
Expanded the "Audio" sections of the White Pages, with additional nodes and information from empirical knowledge.

### CODE REVIEW

File: docs/Audio.md [hunk 1]
```diff
@@ -29,12 +29,62 @@ Determines if a referenced track is currently playing. Requires a valid referenc
...
+
 ### Suspend BGM
 
 Suspends the currently playing BGM. If resumed, will continue from the time it was suspended.
 
+Also exists with an "Optional Fade" variant, which requires a fade duration (in sec.)
+
 ### Resume BGM
 
+Will resume the music where it left off, with a fade in. The fade duration is set in the BGM controller's parameters.
```
Comment (cpazzra) [inline docs/Audio.md]: Maybe specify `where suspended` or something to that effect to make it clear the two nodes are related.

File: docs/Audio.md [hunk 2]
```diff
@@ -29,12 +29,62 @@ Determines if a referenced track is currently playing. Requires a valid referenc
...
+
+Also exists with an "Optional Fade" variant, which requires a fade duration (in sec.) which will override the BGM controller's.
+
+### Get Musical Time
+
+Given a BGM controller, gets the play time of its currently BGM track. The play time's value ("Musical time") is not in seconds, and does not directly scale with real-time values.
```
Comment (cpazzra) [inline docs/Audio.md]: Out of curiosity, what scale is this value in? Game frames/ticks? Or do we not know?

File: docs/Audio.md [hunk 3]
```diff
@@ -29,12 +29,62 @@ Determines if a referenced track is currently playing. Requires a valid referenc
...
+
+#### Converting Musical Time values into real-time
+
+Each BGM controller has a (hard-coded?) range of value for the musical time. This range is determined by the duration of the unmodded BGM track that the BGM controller calls:
+- BGM track has a loop:
+	- Musical time range = -1 to (unmodded BGM track length)/2
+- BGM track doesn't have a loop:
+	- Musical time range = 0 to (unmodded BGM track length)/2
+
+These values **don't** change even when replacing the original BGM track.
```
Comment (cpazzra) [inline docs/Audio.md]: This *sounds* like you can't change where a track loops when replacing audio? Am I reading this right?

**Review by cpazzra — COMMENTED** [review]
Had a few questions, but looks mostly good.

File: docs/Audio.md (hunk 3 above)
Comment (Timoram) [reply docs/Audio.md]: Not quite, you can replace any looped BGM track with a new BGM with a different loop, but the BGM controller will continue to tick Musical Time values based on the unmodded track. I'll update the page!

File: docs/Audio.md (hunk 2 above)
Comment (Timoram) [reply docs/Audio.md]: It's a bit complex, I'll refer to a section for that!

File: docs/Audio.md (hunk 1 above)
Comment (Timoram) [reply docs/Audio.md]: Will do, thanks for the suggestion!

**Review by cpazzra — APPROVED** [review]

**Comment by Timoram** [comment]
@CpazR Don't forget to merge the branch for the changes to appear in the main one!

**Comment by cpazzra** [comment]
> @CpazR Don't forget to merge the branch for the changes to appear in the main one!

Oh! I thought you should be able to. But sure, I can do that.

**Pull request merged**